src/internal/FluencePeer.ts

```diff
@@ -453,22 +415,82 @@ export class FluencePeer {
         };
     }
 
-    // private
+    /**
+     * @private Subject to change. Do not use this method directly
+     */
+    async init(config: PeerConfig & Required<Pick<PeerConfig, 'KeyPair'>>) {
+        if (!config.KeyPair) {
+            throw new Error('Key Pair is required');
+        }
```

Comment:
When you added the type `& Required<Pick<PeerConfig, 'KeyPair'>>` above you removed the necessety for this run time `if (!config.KeyPair)` check. Please remove it